test/e2e/load.go

```diff
@@ -343,69 +361,67 @@ func sleepUpTo(d time.Duration) {
 	time.Sleep(time.Duration(rand.Int63n(d.Nanoseconds())))
 }
 
-func createAllRC(configs []*testutils.RCConfig, creatingTime time.Duration) {
+func createAllResources(configs []testutils.RunObjectConfig, creatingTime time.Duration) {
 	var wg sync.WaitGroup
 	wg.Add(len(configs))
 	for _, config := range configs {
-		go createRC(&wg, config, creatingTime)
+		go createResource(&wg, config, creatingTime)
 	}
 	wg.Wait()
 }
 
-func createRC(wg *sync.WaitGroup, config *testutils.RCConfig, creatingTime time.Duration) {
+func createResource(wg *sync.WaitGroup, config testutils.RunObjectConfig, creatingTime time.Duration) {
 	defer GinkgoRecover()
 	defer wg.Done()
 
 	sleepUpTo(creatingTime)
-	framework.ExpectNoError(framework.RunRC(*config), fmt.Sprintf("creating rc %s", config.Name))
+	framework.ExpectNoError(config.Run(), fmt.Sprintf("creating %v %s", config.GetKind(), config.GetName()))
 }
 
-func scaleAllRC(configs []*testutils.RCConfig, scalingTime time.Duration) {
+func scaleAllResources(configs []testutils.RunObjectConfig, scalingTime time.Duration) {
 	var wg sync.WaitGroup
 	wg.Add(len(configs))
 	for _, config := range configs {
-		go scaleRC(&wg, config, scalingTime)
+		go scaleResource(&wg, config, scalingTime)
 	}
 	wg.Wait()
 }
 
 // Scales RC to a random size within [0.5*size, 1.5*size] and lists all the pods afterwards.
 // Scaling happens always based on original size, not the current size.
-func scaleRC(wg *sync.WaitGroup, config *testutils.RCConfig, scalingTime time.Duration) {
+func scaleResource(wg *sync.WaitGroup, config testutils.RunObjectConfig, scalingTime time.Duration) {
 	defer GinkgoRecover()
 	defer wg.Done()
 
 	sleepUpTo(scalingTime)
-	newSize := uint(rand.Intn(config.Replicas) + config.Replicas/2)
-	framework.ExpectNoError(framework.ScaleRC(config.Client, config.InternalClient, config.Namespace, config.Name, newSize, true),
-		fmt.Sprintf("scaling rc %s for the first time", config.Name))
-	selector := labels.SelectorFromSet(labels.Set(map[string]string{"name": config.Name}))
-	options := v1.ListOptions{
-		LabelSelector:   selector.String(),
-		ResourceVersion: "0",
-	}
-	_, err := config.Client.Core().Pods(config.Namespace).List(options)
-	framework.ExpectNoError(err, fmt.Sprintf("listing pods from rc %v", config.Name))
+	newSize := uint(rand.Intn(config.GetReplicas()) + config.GetReplicas()/2)
+	framework.ExpectNoError(framework.ScaleResource(
+		config.GetClient(), config.GetInternalClient(), config.GetNamespace(), config.GetName(), newSize, true, config.GetKind()),
+		fmt.Sprintf("scaling rc %s for the first time", config.GetName()))
```

Comment:
Please revert the change with List - this was added explicitly to test listing too.